### DISCUSSION TITLE
Distributed install - Missing Grid hosts + DNS Search Domain + OS Patch Update Source Questions

### DISCUSSION THREAD

**Discussion by Sazerak** [Security-Onion-Solutions/securityonion#9929] [Unsupported Versions]
Hi all! Just setup my first distributed install and have a couple of questions. I'm not sure if it would have been more appropriate to make separate topics for these, but maybe some of the items are related, so I'm going with a single post for now.

Let's start with the first question: I have 4 nodes, a manager, a search, and 2 forward nodes. At first I thought my setup wasn't working since I couldn't see any nodes on the Grid page, but eventually I noticed that alerts were coming in, I can see all nodes in Grafana, and running commands like `salt \* test.ping` from the manager gets a response from all nodes. So I'm really puzzled as to why I'm not seeing any of them show up on the Grid. I've rebooted all hosts and checked my firewall rules and everything seems ok.

That brings me to question 2. During setup for each node, I identified a DNS Search Domain (I think it was actually called default search domain) for the domain I use in my network (let's call it `mydomain.lan` for now). Now the issue I'm running into is that I'm seeing a crazy amount of DNS requests for bad searches because the domain name is being repeated. It seems to be all related to the manager's domain name from what I can tell. My manager's FQDN is `nsm.mydomain.lan` and while some DNS requests are actually being done for that domain, most are querying for `nsm.mydomain.lan.mydomain.lan`.

My last question should hopefully be a simple one: During setup, I identified that the various nodes would get their updates Directly, but I've since decided that I want to have updates go through the manager instead so I can limit internet access to the manager only. I found in the individual minion sls files that I can change the setting, but I can't for the life of me remember what the correct option was called during setup. I think it was `direct+proxy` or `direct+manager` but I can't remember and I don't even know if the sls file uses the same format. I've tried searching through Discussions and the Documentation but haven't found the answer.

On that note, if the answer is actually in writing somewhere, if someone could provide a link I would greatly appreciate it. I've also considered simply re-running the ISO setup, but that feels like overkill for such a minor change. Would that be the appropriate way of doing it though?

Thanks in advance for any guidance!

**Comment by InfosecGoon** [comment]
For the first question: make sure that you'd got communication between the minion and Manager nodes on all the required ports. There's a good list of them here: https://docs.securityonion.net/en/2.3/firewall.html

The Grid page is built dynamically when minion nodes check in, so if you're not seeing them there you've probably blocked either port 443 (sensoroni) or port 8086 (influxdb) from your minions to your Manager.

For question two, try adding nsm.mydomain.lan.mydomain.lan to the /etc/hosts file on the minions so they don't need to look it up in DNS.

For question three, change the settings under "patch" in the minion file (/opt/so/saltstack/local/pillar/minions/*.sls)

```
patch:
  os:
    source: 'manager'
    schedule_name: 'auto'
    enabled: True
    splay: 300
```

**Reply by Sazerak** [reply]
Thanks for your reply. The firewall between the nodes and the manager is wide open so it shouldn't be blocking anything. I did notice that I wasn't getting any states showing up which I found odd, but I did run a tcpdump and there was definitely traffic going back and forth between the nodes and the manager. I didn't thoroughly check if I noticed traffic on all the required ports, but I remember seeing at least a few, so I can't imagine the firewall being the issue.

As for the random DNS queries, I don't think that putting a manual entry in the hosts files would be a good idea since that would only mask the issue, not actually fix it. What I really need to find out is why the nodes are repeating the domain name in their request.

**Comment by Sazerak** [comment]
Looks like I've somewhat fixed my first 2 issues. I noticed there was no traffic between the nodes and the manager on 443 and it was actually related to DNS. But to make matters more complicated, it's because I'm using a Virtual IP (in pfSense) to expose the manager to the "outside" world (this is a nested firewall configuration, so outside in this case is still inside my network).

Looks like DNS resolution wasn't happening for my local domain, so I configured it to work, but then ran into the issue that then the nodes were trying to resolve the Virtual IP which wouldn't work, so for now I've set a host override in the firewall and I'll keep checking to see how I can make this work even better.
